library/autodiff/examples/neohookean.rs

```diff
@@ -0,0 +1,363 @@
+    }
+}
+
+pub struct NH {
+    pub mu: f64,
+    pub lambda: f64,
+}
+
+impl NH {
+    pub fn from_lame(lambda: f64, mu: f64) -> Self {
+        Self { lambda, mu }
+    }
+    // https://en.wikipedia.org/wiki/Lam%C3%A9_parameters
+    pub fn from_youngs(E: f64, nu: f64) -> Self {
+        Self {
+            lambda: E * nu / ((1.0 + nu) * (1.0 - 2.0 * nu)),
+            mu: E / (2.0 * (1.0 + nu)),
+        }
+    }
+}
+
+// Helmholtz free energy density
+#[autodiff(d_psi, Reverse, Active)]
+fn psi(#[dup] e: &KM, nh: &NH) -> f64 {
+    let mu = nh.mu;
+    let lambda = nh.lambda;
+    let J = e.cauchy_green().det().sqrt();
+    0.25 * lambda * (J * J - 1.0 - 2.0 * J.ln()) // convex
+      + mu * (e.trace() - J.ln())
+}
+
+// Kinchhoff stress (current configuration)
+#[autodiff(d_stress, Forward, DuplicatedNoNeed)]
+fn stress(#[dup] e: &KM, nh: &NH) -> KM {
+    let mut dpsi_de = KM::zero();
+    d_psi(&e, &mut dpsi_de, &nh, 1.0);
+    let b = e.cauchy_green();
+    dpsi_de * b
+}
+
+pub mod analytic {
+    use super::*;
+    pub fn stress(e: &KM, nh: &NH) -> KM {
+        let lambda = nh.lambda;
+        let mu = nh.mu;
+        let b = e.cauchy_green();
+        let J = b.det().sqrt();
+        let I = KM::identity();
+        0.5 * lambda * (J * J - 1.0) * I + 2.0 * mu * e
+    }
+
+    pub fn d_stress(e: &KM, ddudx: &Mat3x3, nh: &NH) -> KM {
+        let lambda = nh.lambda;
+        let mu = nh.mu;
+        let b = e.cauchy_green();
+        let J = b.det().sqrt();
+        let tau = stress(e, nh).to_matrix();
+        let deps = KM::epsilon(ddudx);
+        let I = KM::identity();
+        let FdSFt = lambda * J * J * deps.trace() * I + (2.0 * mu - lambda * (J * J - 1.0)) * deps;
+        2.0 * KM::from_matrix(matmul(&ddudx, false, &tau, false)) + FdSFt
+    }
+}
+
+fn main() {
+    let nh = NH::from_lame(2.0, 1.0);
+    let e = KM::from_voigt(&[0.2, -0.1, -0.12, 0.3, 0.2, 0.15]);
+    let energy = psi(&e, &nh);
+    println!("energy: {energy}");
+
+    let mut dpsi_de = KM::zero();
+    d_psi(&e, &mut dpsi_de, &nh, 1.0);
+    println!("dpsi_de: {dpsi_de:?}");
+    let stress = stress(&e, &nh);
+    println!("enzyme stress  : {stress:?}");
+    let stress = analytic::stress(&e, &nh);
+    println!("analytic stress: {stress:?}");
+
+    let ddudx = [[1.0, 1.1, 1.2], [1.3, 1.4, 1.5], [1.6, 1.7, 1.8]];
+    let deps = KM::epsilon(&ddudx);
+    if false {
+        // rustc: /home/jeka2967/src/rust/src/llvm-project/llvm/lib/Analysis/ScalarEvolution.cpp:8581: llvm::ScalarEvolution::ExitLimit llvm::ScalarEvolution::computeExitLimitFromSingleExitSwitch(const llvm::Loop*, llvm::SwitchInst*, llvm::BasicBlock*, bool): Assertion `L->contains(Switch->getDefaultDest()) && "Default case must not exit the loop!"' failed.
+        let de = 2.0 * KM::from_matrix(matmul(&ddudx, false, &e.to_matrix(), false)) + deps;
+        let dtau = d_stress(&e, &de, &nh);
+        println!("enzyme dstress  : {dtau:?}");
+    }
```

Comment:
This currently fails to compile.

Reply:
Building, I'll check it in the morning.

Reply:
Actually, I have this problem on branch `rebase`. Do I need different options to configure there?
```console
$ ../configure --enable-llvm-link-shared --enable-llvm-plugins --enable-llvm-enzyme --release-channel=nightly --enable-llvm-assertions --enable-clang --enable-lld --enable-option-checking --enable-ninja --disable-docs
configure: processing command line
configure:
configure: build.configure-args := ['--enable-llvm-link-shared', '--enable-llvm-p ...
configure: llvm.link-shared     := True
configure: llvm.plugins         := True
configure: llvm.enzyme          := True
configure: rust.channel         := nightly
configure: llvm.assertions      := True
configure: llvm.clang           := True
configure: rust.lld             := True
configure: llvm.ninja           := True
configure: build.docs           := False
configure: profile              := user
Traceback (most recent call last):
  File "/home/jeka2967/src/rust/build/../src/bootstrap/configure.py", line 549, in <module>
    section_order, sections, targets = parse_args(sys.argv[1:])
                                       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jeka2967/src/rust/build/../src/bootstrap/configure.py", line 279, in parse_args
    return parse_example_config(known_args, config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jeka2967/src/rust/build/../src/bootstrap/configure.py", line 433, in parse_example_config
    configure_file(sections, top_level_keys, targets, config)
  File "/home/jeka2967/src/rust/build/../src/bootstrap/configure.py", line 509, in configure_file
    configure_section(sections[section_key], section_config)
  File "/home/jeka2967/src/rust/build/../src/bootstrap/configure.py", line 485, in configure_section
    raise RuntimeError("failed to find config line for {}".format(key))
RuntimeError: failed to find config line for enzyme
```